wolfcrypt/src/tfm.c

```diff
@@ -643,19 +671,26 @@ int fp_div(fp_int *a, fp_int *b, fp_int *c, fp_int *d)
   /* normalize both x and y, ensure that y >= b/2, [b == 2**DIGIT_BIT] */
   norm = fp_count_bits(y) % DIGIT_BIT;
   if (norm < (int)(DIGIT_BIT-1)) {
-     norm = (DIGIT_BIT-1) - norm;
-     fp_mul_2d (x, norm, x);
-     fp_mul_2d (y, norm, y);
+    int ret;
+    norm = (DIGIT_BIT-1) - norm;
+    ret = fp_mul_2d (x, norm, x);
+    if (ret != FP_OKAY) {
+    #ifdef WOLFSSL_SMALL_STACK
+      XFREE(q, NULL, DYNAMIC_TYPE_BIGINT);
+    #endif
+      return ret;
+    }
+    (void)fp_mul_2d (y, norm, y);
```

Comment:
Why wouldn't we check the return codes for the following functions that can now fail?

Reply:
In this case, the code is moving the top bit of y up to the maximum bit in the digit.
The function can't fail.

Reply:
What about the 14 cases below as well?  In each case, we need to document why (ideally with a proof) that it can never fail.  A scanner probably won't know this or a future reader of the code.

Reply:
Put in checks everywhere I could find.
Multiple functions now return error codes and put in checks where they are called as well.